src/transformers/JSXTransformer.ts

```diff
@@ -83,14 +83,28 @@ export default class JSXTransformer extends Transformer {
 
   processProps(firstTokenStart: number): void {
     const lineNumber = this.getLineNumberForIndex(firstTokenStart);
-    const devProps = `__self: this, __source: {fileName: ${this.getFilenameVarName()}, lineNumber: ${lineNumber}}`;
+    const devProps = this.options.production
+      ? ""
+      : `__self: this, __source: {fileName: ${this.getFilenameVarName()}, lineNumber: ${lineNumber}}`;
     if (!this.tokens.matches1(tt.jsxName) && !this.tokens.matches1(tt.braceL)) {
-      this.tokens.appendCode(`, {${devProps}}`);
+      if (devProps) {
+        this.tokens.appendCode(`, {${devProps}}`);
+      } else {
+        this.tokens.appendCode(`, null`);
+      }
       return;
     }
     this.tokens.appendCode(`, {`);
+    let firstProperty = true;
+    const appendComma = () => {
```

Comment:
Any ideas how to make this cleaner?

Reply:
Is the `appendComma` change just there to avoid the trailing comma? It's fine to assume that any compilation target supports trailing object commas, so I'd prefer to just keep the trailing comma in the output for simplicity.

The original code before I added `devProps` was to just always have a trailing comma: https://github.com/alangpierce/sucrase/pull/176/files#diff-17d5ba94e1a4d345c1187a5cb0b54ad7R96

Also, the closure approach here is unlikely to work with AssemblyScript (which currently doesn't work on all of Sucrase, but I'd like to move in a direction where it eventually can), so simpler control flow would be better here.